crates/swc_ecma_minifier/src/compress/optimize/iife.rs

```diff
@@ -582,6 +592,10 @@ impl Optimizer<'_> {
                 trace_op!("iife: Expr::Fn(..)");
 
                 if !self.may_add_ident() {
```

Comment:
@Austaras This condition can be `!false` if it's a top-level, but the previous code added variables to the top-level scope because we didn't check for parameter count